### PR TITLE
Add MaskRCNN model as a test for the wrapper approach with TF model garden code

### PR DESCRIPTION
**Pull request by qlzh727**
# Note that this PR is more for a demo and not intend to be submitted at the moment. 

# What does this PR do?
Add MaskRCNN model as a test for the wrapper approach with TF model garden code.

1. The current class only contains the model building logic.
2. Most of the tunable params are currently backed in the function, and not exposed to use yet.
3. Verify the model building logic in the test. (will need tfm pip deps to work)

Next step:
1. Configure the input parsing logic for coco, and also anchor generation.
2. Mimic the training loop logic as model.train_step/eval_step
3. Train model e2e.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

### CODE REVIEW

**Comment by innat** [comment]
Relevant ticket. https://github.com/keras-team/keras-cv/issues/623

**Review by martin-gorner — CHANGES_REQUESTED** [review]
Thank you for this first attempt. 
To bring this further, could you be able to test it actually training, with compile and fit working ?

I have created a starter MaskRCNN notebook here with a small bur real dataset that can be used for further testing:
https://source.cloud.google.com/cloudml-demo-martin/martin-kerascv-test/+/master:Mask_RCNN_test.ipynb
I have not been able to test very far though.

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        # Code before yield runs before the test
+        yield
+        tf.keras.backend.clear_session()
+
+    def test_maskrcnn_construction(self):
+        mask_rcnn_model = keras_cv.models.mask_rcnn(
+            classes=91,     # default value for coco
+            input_size=[512, 512, 3],
+            backbone="resnet",
+            backbone_weights=None,
```
[the remark this answers is not in the document]
Comment (bhack) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py:35]: Currently `supported_premade_backbone = ['resnet', 'spinenet', 'mobilenet']ì

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @pytest.fixture(autouse=True)
+    def cleanup_global_session(self):
+        # Code before yield runs before the test
+        yield
+        tf.keras.backend.clear_session()
+
+    def test_maskrcnn_construction(self):
+        mask_rcnn_model = keras_cv.models.mask_rcnn(
+            classes=91,     # default value for coco
+            input_size=[512, 512, 3],
```
Comment (martin-gorner) [inline keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py:33]: Is the input_size parameter necessary ?
Usually in Keras models, this gets built from the data.

File: keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (martin-gorner) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py:35]: I tried "spinenet" but it errors out:

decoder = tfm.vision.decoders.FPN(
--> 129         input_specs=backbone.output_specs,  # Might need to check for custom backbones
    130         min_level=min_level,
    131         max_level=max_level,

AttributeError: 'str' object has no attribute 'output_specs'

[the remark this answers is not in the document]
Comment (bhack) [reply keras_cv/models/object_detection/mask_rcnn/mask_rcnn_test.py:35]: Yes these backends are TODO only resnet now:

`https://github.com/keras-team/keras-cv/pull/754/files#diff-e505a34eb6530028aa2ea9a075609b26d3aaa4811eef0179d40d4eab04bf056aR112-R114

**Comment by martin-gorner** [comment]
Open questions:
 - how should loss configuration work ? There are three losses in MaskRCNN: classification loss, box regression loss and mask loss. An additional complexity id that masks are predicted per class but only the mask loss of the predicted class is taken into account.
 - metric configuration: this should be compatible with KerasCV's COCO metrics.
 - support for KerasCV-style bounding box formats.
 - TPU support ? 
 - can data encoding/decoding use KerasCV APIs?

**Comment by tanzhenyu** [comment]
Should we close this as we most likely will add MaskRCNN on top of existing FasterRCNN?

**Comment by LukeWood** [comment]
> Should we close this as we most likely will add MaskRCNN on top of existing FasterRCNN?

yeah I think so.

**Comment by tanzhenyu** [comment]
Discussed offline, it seems we all agree to close it for now.

**Pull request closed without merging**